### PR TITLE
[Non_Sanity_Testing] Fixed non-sanity test case for ESM info timer expiration

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
[Non_Sanity_Testing] Fixed non-sanity test case for ESM info timer expiration

## Summary
This PR is partial fix for a subset of issues from github issue #5842 (Also refer PR #5840 which adds all these non-sanity test cases as part of defs.mk file)

The non-sanity test case **s1aptests/test_attach_esm_info_timerexpiration_max_retries.py** is getting stuck and does not lead to failure or success. The problem is that the test case is waiting for ESM information request 4 times before sending the ESM information answer, however, as per spec and MME implementation, the maximum no. of retransmissions allowed is 3. Also, since MME does not send the PDN connectivity reject message as of now after maximum number of T3489 expires, test script does not receive any failure response and has no provision to handle it as well. Therefore, test case gets stuck waiting for response, if num_of_expires value is greater than 3. This PR fixes the issue by reducing the num_of_expires to 3 to match with the current implementation in MME.

The test case was getting stuck at this place:
```
(python) vagrant@magma-test:~/magma/lte/gateway/python/integ_tests$ make integ_test TESTS=s1aptests/test_attach_esm_info_timerexpiration_max_retries.py. /home/vagrant/build/python/bin/activate
echo "Running test: s1aptests/test_attach_esm_info_timerexpiration_max_retries.py"
Running test: s1aptests/test_attach_esm_info_timerexpiration_max_retries.py
timeout --foreground -k 930s 900s sudo -E PATH=/home/vagrant/build/python/bin:/usr/lib/ccache:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin PYTHONPATH=:/home/vagrant/s1ap-tester/bin /home/vagrant/build/python/bin/nosetests --with-xunit --xunit-file=/var/tmp/test_results/test_attach_esm_info_timerexpiration_max_retries.xml -x -s s1aptests/test_attach_esm_info_timerexpiration_max_retries.py || exit 1
Start time 11:31:25
tag: AUTH_TYPE tagVal: MILENAGE
************************* Enb tester config

**********************TPT open server******** 0

 Server Socket FD =[13]
************* Testing *Time elapsed(in usec): for timer expiry               :4084 usec
AGW is stateless
APN Correction configured
Health service is disabled
Using subscriber IMSI IMSI001010000000001
************************* UE device config for ue_id  1
************************* Configuring IP block
************************* Waiting for IP changes to propagate
************************* S1 setup
************************* UE App config
************************* sending Attach Request for ue-id : 1
Sending Attach Request ue-id 1
Received auth req ind
Sending Auth Response ue-id 1
Received Security Mode Command ue-id 1
Num of expires =  4
Received Esm Information Request ue-id 1
Got the response
Received Esm Information Request ue-id 1
Got the response
Received Esm Information Request ue-id 1
Got the response
Received Esm Information Request ue-id 1
```

## Test Plan
Verified after running the test cases multiple time.

The terminal success log with this PR changes from S1APTester VM has been attached here for reference:
[s1aptester_terminal_log.txt](https://github.com/magma/magma/files/6269728/s1aptester_terminal_log.txt)

 

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

File: lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py [hunk 1]
```diff
@@ -31,7 +30,13 @@ def tearDown(self):
     def test_esm_information_timerexpiration(self):
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py]: can you modify the test name in line with the file name, e.g., `test_esm_info_timerexpiration_max_retries`?

File: lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py [hunk 2]
```diff
@@ -31,7 +30,13 @@ def tearDown(self):
...
         """ Testing of sending Esm Information procedure """
         num_ues = 1
-        num_of_expires = 4
+        # The maximum no. of allowed transmissions of ESM information request
+        # as per 3gpp spec 24.301 is 3. Since MME does not send the PDN
+        # connectivity reject message as of now after maximum number of
+        # T3489 expires, test script does not receive any failure response and
+        # has no provision to handle it as well. Therefore, test case gets
+        # stuck waiting for response, if num_of_expires value is greater than 3
+        num_of_expires = 3
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py]: change name to max_retries

**Review by ulaskozat — APPROVED** [review]
there are a few nits in names and variable names, could you fix them while at this?

File: lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py (hunk 1 above)
Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py]: Done

File: lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py (hunk 2 above)
Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/test_attach_esm_info_timerexpiration_max_retries.py]: Done

**Comment by VinashakAnkitAman** [comment]
> there are a few nits in names and variable names, could you fix them while at this?

I have updated the PR with the updated variable/class/function names as per the review comments

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**